db/dbtypes/types.go

```diff
@@ -975,11 +973,12 @@ func MergeRows(rows []*AddressRow) ([]AddressRowMerged, error) {
 				mr.AtomsDebit = r.Value
 			}
 
-			merged[*Hash] = &mr
+			hashMap[*Hash] = &mr
+			mergedRows = append(mergedRows, &mr)
 			continue
 		}
 
-		// Update existing transaction.
+		// Update existing transaction in the mergedRows slice.
```

Comment:
`row` comes out of the `hashMap`, not the slice, but the struct it points to is the same.